cli/tools/test.rs

```diff
@@ -285,8 +286,13 @@ impl PrettyTestReporter {
       colors::gray(format!("({})", display::human_elapsed(elapsed.into())))
     );
 
-    if let Some(error_text) = result.error() {
-      for line in error_text.lines() {
+    if let Some(js_error) = result.error() {
+      // TODO(bartlomieju): use structured data here
+      let err_string = js_error
+        .to_string()
+        .trim_start_matches("Uncaught ")
+        .to_string();
```

Comment:
I think we should implement the `JsError::aggregate` printing in `PrettyJsError` (addressing #13831) and use that formatting here.
```suggestion
      let err_string = PrettyJsError::create(js_error)
        .to_string()
        .trim_start_matches("Uncaught ")
        .to_string();
```
Using our colourful terminal error formatting would be good for test failures. #14189 can still be done on top.

`JsError::fmt()` should remain as it is (a simple implementation around `write!(f, "{}", self.stack)`) rather than being a third formatting implementation that has to be mindful of test output.

Reply:
@nayeemrmn shouldn't we do both? What I implemented here is not really specific to tests, it's just a way to print aggregate error without multiple "Uncaught" in the middle. I agree we should implement it for `PrettyJsError`, but not implementing it for `JsError` means that the only information you'll get will be `AggregateError` if it's not formatted by `PrettyJsError` later. 

Anyway, I'll look into doing `PrettyJsError` first.

Reply:
Reopened https://github.com/denoland/deno/pull/14285, but it suffers from the same problem of having to manually trim "Uncaught". Please take a look and let me know if that approach looks fine for you.

Reply:
> What I implemented here is not really specific to tests

No I mean that this would become the only place that `JsError::fmt()` is exposed in Deno cli. I believe currently it's only there for consumers of core or debugging. IMO this code here should definitely defer to `PrettyJsError`, and `JsError::fmt()` should probably be kept minimal until there is some use/demand from core users, otherwise it is a third error formatting implementation to maintain.

Reply:
Sure, I'll remove that implementation

Reply:
Addressed in https://github.com/denoland/deno/pull/14285